readthedocs_build/config/test_find.py

```diff
@@ -75,3 +75,18 @@ def test_find_multiple_files(tmpdir):
         str(tmpdir.join('first', 'readthedocs.yml')),
         str(tmpdir.join('third', 'readthedocs.yml')),
     ]
+
+
+def test_find_unicode_path(tmpdir):
+    base_path = os.path.abspath('integration_tests/bad_encode_project')
+    try:
+        path = find_one(base_path, ('readthedocs.yml',))
+        # When python3
+        assert path == ''
+    except Exception as e:
+        # When python2
+        assert isinstance(e, UnicodeDecodeError)
+
+        unicode_base_path = base_path.decode('utf-8')
```

Comment:
So the final question, should we force always to be `str` on the `find_one` method or we should document that anyone calling this function must pass a `str` object and not an `unicode`?

Reply:
Also probably would be better to explicitly check the python version on the test, right?

Reply:
We need to be able to support UTF8 file names, so I think we just need to patch the walk method. We did this somewhere -- either readthedocs.org or sphinx-autoapi (i think)